src/MongoDB.Driver.Core/Core/Operations/FindCommandOperation.cs

```diff
@@ -484,10 +484,11 @@ private AsyncCursor<TDocument> CreateCursor(IChannelSourceHandle channelSource,
         {
             var getMoreChannelSource = new ServerChannelSource(channelSource.Server, channelSource.Session.Fork());
             var firstBatch = CreateCursorBatch(commandResult);
+            var cursorDocument = commandResult["cursor"].AsBsonDocument;
 
             return new AsyncCursor<TDocument>(
                 getMoreChannelSource,
-                _collectionNamespace,
+                CollectionNamespace.FromFullName(cursorDocument["ns"].AsString),
```

Comment:
With the above refactoring this line could simply be:

    collectionNamespace,

I'd like to see us moving away from doing work inside a function call. It increases complexity and makes it harder to step into a function when using the debugger.

Reply:
Done.